components/formats-bsd/src/loci/formats/out/TiffWriter.java

```diff
@@ -499,4 +545,74 @@ else if (retrieve.getPixelsBinDataCount(series) == 0) {
     tiffSaver.setCodecOptions(options);
   }
 
+  @Override
+  public int getTileSizeX() throws FormatException {
+    if (tileSizeX == 0) {
+      tileSizeX = super.getTileSizeX();
+    }
+    return tileSizeX;
+  }
+
+  @Override
+  public int setTileSizeX(int tileSize) throws FormatException {
+    tileSizeX = super.setTileSizeX(tileSize);
+    tileSizeX = Math.round((float)tileSize/TILE_GRANULARITY) * TILE_GRANULARITY;
```

Comment:
The floating point was used to round to the nearest value, if we were only every to round upwards (which I would be fine with) then it could be removed.